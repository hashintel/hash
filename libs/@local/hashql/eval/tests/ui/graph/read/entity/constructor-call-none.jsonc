//@ run: pass
//@ description: Test constructor calls with None type in filter expressions
// biome-ignore format: readability
["::graph::tail::collect",
  ["::graph::body::filter",
    ["::graph::head::entities", ["::graph::tmp::decision_time_now"]],
    ["fn", { "#tuple": [] }, { "#struct": { "vertex": "_" } }, "_",
      ["==",
        ["None"],
        ["None"]
      ]
    ]
  ]
]
