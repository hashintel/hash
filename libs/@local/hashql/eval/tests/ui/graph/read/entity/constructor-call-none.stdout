════ HIR ═══════════════════════════════════════════════════════════════════════

::core::graph::head::entities(::core::graph::tmp::decision_time_now())
|> ::core::graph::body::filter(#fn(vertex:0: _0«Entity[(id:
 EntityRecordId[(edition_id: EntityEditionId[Uuid[String]], entity_id:
  EntityId[(draft_id: (None[Null] | Some[DraftId[Uuid[String]]]), entity_uuid:
   EntityUuid[Uuid[String]], web_id:
   WebId[ActorGroupEntityUuid[Uuid[String]]])])], link_data: (None[Null]
  | Some[LinkData[(left_entity_id: EntityId[(draft_id: (None[Null]
     | Some[DraftId[Uuid[String]]]), entity_uuid: EntityUuid[Uuid[String]],
    web_id: WebId[ActorGroupEntityUuid[Uuid[String]]])], right_entity_id:
   EntityId[(draft_id: (None[Null] | Some[DraftId[Uuid[String]]]), entity_uuid:
    EntityUuid[Uuid[String]], web_id:
    WebId[ActorGroupEntityUuid[Uuid[String]]])])]]), properties: ?)]»):
_1«Boolean» ->
    (#ctor(fn() -> None[Null], arguments: [])() == #ctor(fn() -> None[Null],
    arguments: [])())) |> ::core::graph::tail::collect

════ Entity Filter ═════════════════════════════════════════════════════════════

[
    Equal(
        Some(
            Parameter {
                parameter: Any(
                    Null,
                ),
                convert: None,
            },
        ),
        Some(
            Parameter {
                parameter: Any(
                    Null,
                ),
                convert: None,
            },
        ),
    ),
]
